src/Compilers/Test/Core/Compilation/ControlFlowGraphVerifier.cs

```diff
@@ -1604,30 +1657,54 @@ void recordReferences(IOperation operation)
                 {
                     IMethodSymbol method;
 
-                    switch (node.Kind)
+                    switch (node)
                     {
-                        case OperationKind.LocalReference:
-                            referencedLocalsAndMethods.Add(((ILocalReferenceOperation)node).Local);
+                        case ILocalReferenceOperation localReference:
+                            if (localReference.Local.ContainingSymbol.IsTopLevelMainMethod() && !associatedSymbol.IsTopLevelMainMethod())
```

Comment:
>associatedSymbol [](start = 97, length = 16)

Perhaps we should assert somewhere that ```associatedSymbol``` is not a local function or a lambda, otherwise the condition might not be accurate.

Reply:
It's fine if it's a local function or lambda, but we likely want to skip only if that local function or lambda isn't itself part of the top-level main.

---
In reply to: [529792485](https://github.com/dotnet/roslyn/pull/49534#discussion_r529792485) [](ancestors = 529792485)